src/test/java/bio/overture/ego/controller/PolicyControllerTest.java

```diff
@@ -0,0 +1,295 @@
+  }
+
+  @Test
+  @SneakyThrows
+  public void addUniquePolicy() {
+    val policy1 = Policy.builder().name("PolicyUnique").build();
+    val policy2 = Policy.builder().name("PolicyUnique").build();
+
+    val entity1 = new HttpEntity<Policy>(policy1, headers);
+    val response1 =
+        restTemplate.exchange(
+            createURLWithPort("/policies"), HttpMethod.POST, entity1, String.class);
+
+    val responseStatus1 = response1.getStatusCode();
+    assertThat(responseStatus1).isEqualTo(HttpStatus.OK);
+
+    val entity2 = new HttpEntity<Policy>(policy2, headers);
+    val response2 =
+        restTemplate.exchange(
+            createURLWithPort("/policies"), HttpMethod.POST, entity2, String.class);
+    val responseStatus2 = response2.getStatusCode();
+    assertThat(responseStatus2).isEqualTo(HttpStatus.CONFLICT);
+  }
+
+  @Test
+  @SneakyThrows
+  public void getPolicy() {
+    val policyId = policyService.getByName("Study001").getId();
+    val entity = new HttpEntity<String>(null, headers);
+    val response =
+        restTemplate.exchange(
+            createURLWithPort(String.format("/policies/%s", policyId)),
+            HttpMethod.GET,
+            entity,
+            String.class);
+
+    val responseStatus = response.getStatusCode();
+    val responseJson = MAPPER.readTree(response.getBody());
+
+    assertThat(responseStatus).isEqualTo(HttpStatus.OK);
+    assertThat(responseJson.get("name").asText()).isEqualTo("Study001");
+  }
+
+  @Test
+  @SneakyThrows
+  public void addGroupPermission() {
+    val policyId = entityGenerator.setupSinglePolicy("AddGroupPermission").getId().toString();
+    val groupId = entityGenerator.setupGroup("GroupPolicyAdd").getId().toString();
+
+    val entity = new HttpEntity<String>("WRITE", headers);
+    val response =
+        restTemplate.exchange(
+            createURLWithPort(String.format("/policies/%s/permission/group/%s", policyId, groupId)),
+            HttpMethod.POST,
+            entity,
+            String.class);
+
+    val responseStatus = response.getStatusCode();
+    assertThat(responseStatus).isEqualTo(HttpStatus.OK);
+    // TODO: Fix it so that POST returns JSON, not just random string message
+
+    val getResponse =
+        restTemplate.exchange(
+            createURLWithPort(String.format("/policies/%s/groups", policyId)),
+            HttpMethod.GET,
+            new HttpEntity<String>(null, headers),
+            String.class);
+
+    val getResponseStatus = getResponse.getStatusCode();
+    val getResponseJson = MAPPER.readTree(getResponse.getBody());
+    val groupPermissionJson = getResponseJson.get(0);
+
+    assertThat(getResponseStatus).isEqualTo(HttpStatus.OK);
+    assertThat(groupPermissionJson.get("id").asText()).isEqualTo(groupId);
+    assertThat(groupPermissionJson.get("mask").asText()).isEqualTo("WRITE");
+  }
+
+  @Test
+  @SneakyThrows
+  public void deleteGroupPermission() {
```

Comment:
rename test to `disassociatePermissionsFromGroup_EntitiesAndRelationshipsExisting_Success`